pkgs/development/libraries/science/math/cudnn/generic.nix

```
@@ -94,6 +94,7 @@ in
     # Without --add-needed autoPatchelf forgets $ORIGIN on cuda>=8.0.5.
     postFixup = strings.optionalString (strings.versionAtLeast versionTriple "8.0.5") ''
       patchelf $out/lib/libcudnn.so --add-needed libcudnn_cnn_infer.so
+    patchelf $out/lib/libcudnn_ops_infer.so --add-needed libcublas.so.11 --add-needed libcublasLt.so.11
```

Ahh I may have spoke too soon in making this suggestion... I fear that this won't work with cuda versions != 11... we should test it

---

I wonder if
```
patchelf $out/lib/libcudnn_ops_infer.so --add-needed libcublas.so --add-needed libcublasLt.so
```
would work following the style of the preceding line

---

Also, is cublas always available? Doesn't seem so from here:

https://github.com/NixOS/nixpkgs/blob/e1c91b230122dceca76e600aae5a97d2f590ebb2/pkgs/development/libraries/science/math/cudnn/generic.nix#L9

---

I'm not too worried about that... IIUC cublas comes included in cudatoolkit for versions <=10. but we should just test it on cudaPackages_10 and find out

---

attempted to test on `cudaPackages_10` but ran into https://github.com/NixOS/nixpkgs/issues/245218

---

Ok just tested on `cudaPackages_12` and indeed it fails. However, this fix works on both 11 and 12:
```suggestion
      patchelf $out/lib/libcudnn_ops_infer.so --add-needed libcublas.so --add-needed libcublasLt.so
```